File: src/main/java/org/gangel/graphomance/usecases/node/CreateSingleVertexStringUniqueHashIndex.java
package org.gangel.graphomance.usecases.node;

import org.gangel.graphomance.DbType;
import org.gangel.graphomance.IndexType;

public class CreateSingleVertexStringUniqueHashIndex extends CreateSingleVertexBase {

    public CreateSingleVertexStringUniqueHashIndex() {
        super("Create node with string unique hash index", IndexType.HASH_UNIQUE, false, true);
    }

    @Override
    public boolean skipFor(DbType dbType) {
        return (dbType == DbType.NEO4J);
    }
}

File: src/main/kotlin/org/gangel/graphomance/usecases/node/CreateSingleVertexStringUniqueHashIndex.kt
package org.gangel.graphomance.usecases.node

import org.gangel.graphomance.api.DbType
import org.gangel.graphomance.api.IndexType

class CreateSingleVertexStringUniqueHashIndex :
	CreateSingleVertexBase("Create node with string unique hash index",
						   IndexType.HASH_UNIQUE,
						   false,
						   true) {
	override fun skipFor(dbType: DbType): Boolean {
		return dbType == DbType.NEO4J
	}
}